### DISCUSSION TITLE
Display log timestamp in a different timezone

### DISCUSSION THREAD

**Discussion by iprastha** [opcodesio/log-viewer#220] [Q&A] [answered]
By default, laravel logs everything in UTC, which is best practice. However, is there anyway we could view the log entries under a different timezone? this is specifically to convert the time shown in the "Time" column of the log tables.

**Comment by arukompas** [comment]
Hey @iprastha , it already uses your app's default time zone.

Check the `timezone` option in your `config/app.php` configuration file. On a new Laravel project it's set to UTC.

**Reply by davidsalazar** [reply]
It's bad practice to set your project's timezone to anything other than UTC. Most apps put the timezone in the users table and format everything based on that timezone id. So what @iprastha asked for is valid. I also can't stand looking at my logs in UTC lol.

**Reply by iprastha** [reply]
@davidsalazar , that's on point.
An idea would be like to have an config option for log-viewer (example: "display_timezone"), then uses that setting in the view to convert stored timestamp from laravel timezone setting to log-viewer's "display_timezone" setting.
Would like to know if this implementation is possible without causing issues on performance

**Reply by arukompas** [reply]
Valid points, thanks for sharing! I'll have a look at it tomorrow, should be a quick implementation.

**Reply by arukompas** [reply]
Hey @davidsalazar @iprastha - I've tagged a new release, `v2.4.0` with the ability to set a time zone for the Log Viewer specifically.

Just add this to your `config/log-viewer.php` and set it to one of the [PHP supported time zones](https://www.php.net/manual/en/timezones.php):
```php
    'timezone' => '<name-of-supported-timezone>',
```

**Comment by davidsalazar** [comment]
Thanks I’ll try it out tomorrow.

**Comment by iprastha** [comment, marked as answer]
Thanks @arukompas , i've tested it and it works as expected.
Steps:
1. Update package in composer
```
composer update "opcodesio/log-viewer"
```
2. backup my config/log-viewer.php to config/log-viewer-backup.php
3. Re-publish config file
```
php artisan vendor:publish --tag="log-viewer-config"
```
4. Compare and update config/log-viewer as required
5. Set 'timezone' in log-viewer config to other than null, example: 'Asia/Jakarta'
6. Refresh the log-viewer url in the browser and see the timestamp updated as expected
